clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -986,6 +992,51 @@ def CmpOp : CIR_Op<"cmp", [Pure, SameTypeOperands]> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// ComplexExtractOp
+//===----------------------------------------------------------------------===//
+
+def ComplexComponent_Real : I32EnumAttrCase<"real", 1>;
+def ComplexComponent_Imag : I32EnumAttrCase<"imag", 2>;
+
+def ComplexComponent : I32EnumAttr<
+    "ComplexComponent", "complex number component",
+    [ComplexComponent_Real, ComplexComponent_Imag]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def ComplexExtractOp : CIR_Op<"complex.extract", [Pure]> {
+  let summary = "Extract a specified part of a complex number.";
+  let description = [{
+    `cir.complex.extract` operation reads a component of the operand complex
+    number value.
+
+    The parameter `$component` specifies which component to extract. If it is
+    `real`, the real part of the complex number will be extracted. If it is
+    `imag`, the imaginary part of the complex number will be extracted.
+
+    Example:
+
+    ```mlir
+    !fcmp = !cir.complex<!cir.float>
```

Comment:
`!fcmp` might sounds like "float comparison", perhaps use `!complex` to make the docs more readable.

Reply:
Updated.